YSP-509: Add Code Style to WYSIWYG

laura-johnson opened the pull request:
## [YSP-509: Add Code Style to WYSIWYG](https://yaleits.atlassian.net/browse/YSP-509)

### Description of work
- Adds the ‘Code’ and ‘Code Block’ buttons to the CKEditor toolbar for the ‘Basic HTML’ text formatter with supporting styles
- Adds a demo of the code block in Storybook in the text molecule component.
- Adds highlight.js to the codeblock to highlight the code.

### Functional testing steps:
- [ ] Edit a text block in layout builder that uses the Basic HTML format, add a code snippet or block of code. For the block of code the default is plain text, there is a pull-down arrow on the right of that button that you can use to select from a number of code styles
- [ ] Verify they look as expected (current style shown below)
- [ ] Go to the text molecule in Storybook, verify you see a styled code block at the bottom

<img width="946" alt="highlight-codeblock" src="https://github.com/yalesites-org/yalesites-project/assets/1652730/6b946be0-070f-4b82-a72e-988f389642e1">


Depends on: 
https://github.com/yalesites-org/yalesites-project/pull/655
https://github.com/yalesites-org/atomic/pull/234

### Notes
- I tried to import the CSS from the highlightjs library but webpack doesn't like that. So I grabbed the CSS and put it into the component SCSS. I'm not sure if there is a better way.
- There are a number of CSS styles to choose from for the highlighting. I chose one that I saw in a demo that looked good called "monokai-sublime". Not sure if the group would like to play around with the styles, you can do that here: https://highlightjs.org/demo
- I took my best guess at the different languages we might want. There are many that we could optionally include.

codechefmarc reviewed (approved):
All of this looks good, but would love to get Joe's eyes on it as this is more in his realm. I tested the new styling and it looks great! There is one small change, but not a blocker. There's an extra semicolon in the `typography.stories.js` at the end that shows up in Storybook. I don't think that was you, but while we've got this PR open, may as well remove that:
![Screenshot 2024-05-24 at 1 58 24 PM](https://github.com/yalesites-org/component-library-twig/assets/107938318/6a577ef1-25f9-4a48-835b-c7af06db8a33)

joetower commented on components/01-atoms/typography/text/_yds-text.scss:
```diff
@@ -6,6 +6,95 @@ p {
+    Monokai Sublime style. Derived from Monokai by noformnocontent http://nn.mit-license.org/
+    */
+
+    &.hljs {
+      background: #23241f;
+      color: #f8f8f2;
+      overflow-x: auto;
+    }
```
@laura-johnson I'd like to see all of these hex colors replaced with CSS variables or color tokens, if possible. Adding the colors as tokens is likely more involved than what's needed here, so maybe we could define the colors as CSS variables at the top of this file. 

Something like: 
```
// Code highlighting color variables

:root {
--color-hljs-text: #f8f8f2;
--color-hljs-background: #23241f;
}
```

And then, this part: 
```
   &.hljs {
      background: #23241f;
      color: #f8f8f2;
      overflow-x: auto;
    }
```

Would turn into: 

```
   &.hljs {
      background: var(--color-hljs-text);
      color: var(--color-hljs-background);
      overflow-x: auto;
    }
```

Please let me know if you'd rather tag me in to make these hex to CSS variable changes. 👍

joetower commented on components/01-atoms/typography/text/_yds-text.scss:
```diff
@@ -6,6 +6,95 @@ p {
+  // Code blocks.
+  pre code {
+    display: block;
+    padding: 1em;
```
@laura-johnson This `1em` padding should use a spacing variable. Could you please change this from `padding: 1em` to `padding: var(--size-spacing-5)`?

laura-johnson replied:
Ok great, thanks Joe! I'll make these changes.

laura-johnson commented:
> All of this looks good, but would love to get Joe's eyes on it as this is more in his realm. I tested the new styling and it looks great! There is one small change, but not a blocker. There's an extra semicolon in the `typography.stories.js` at the end that shows up in Storybook. I don't think that was you, but while we've got this PR open, may as well remove that: ![Screenshot 2024-05-24 at 1 58 24 PM](https://private-user-images.githubusercontent.com/107938318/333735007-6a577ef1-25f9-4a48-835b-c7af06db8a33.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1MDU4OTgsIm5iZiI6MTcxNzUwNTU5OCwicGF0aCI6Ii8xMDc5MzgzMTgvMzMzNzM1MDA3LTZhNTc3ZWYxLTI1ZjktNGE0OC04MzViLWM3YWYwNmRiOGEzMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwNFQxMjUzMThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZTZiYTMxNDMwNjZhNjQyYTFjZmQ1MGI1MzUxOWM5NjFmZThjZjlmZTdiZDJmYTlkNmFlNWMyN2ZmYTE3NWViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.QEsrUzd68CTPBd1VEjda0Stm42_YoQ-X7w-E0Jfe3co)

I'll fix that semicolon, thanks @codechefmarc!

laura-johnson commented:
@codechefmarc @joetower I revised the SCSS as per Joe's comments and also added all of the languages requested by the team. And I removed the errant semicolon!

joetower reviewed (approved):
@laura-johnson The updated look great! Thank you!
